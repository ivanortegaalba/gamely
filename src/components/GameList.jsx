import React from 'react';
import PropTypes from '../PropTypes';
import { FlexGrid, Grid } from '../style/styled-components/objects/Grid';
import Game from './Game';

// TODO: Create style-component List
export function GameList({
  games,
  open,
  addToFavourites,
  removeFromFavourites,
}) {
  const Wrapper = open ? Grid : FlexGrid;

  return (
    <Wrapper>
      {games.map((game, idx) => (
        <Game
          key={idx}
          game={game}
          addToFavourites={addToFavourites}
          removeFromFavourites={removeFromFavourites}
        />
      ))}
    </Wrapper>
  );
}

GameList.propTypes = {
  games: PropTypes.arrayOf(PropTypes.game),
};
GameList.defaultProps = {
  open: false,
  games: [],
};

export default GameList;
