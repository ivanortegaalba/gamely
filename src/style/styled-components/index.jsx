export { theme } from './theme'

// Components
export { Header } from './components/Header'
export { Button } from './components/Button'

// Elements
export {H1, H2, H3, H4, H5, H6} from './elements/Headings'

// Settings
export { colors } from './settings/colors'
export { fonts } from './settings/fonts'

// Tools
export { px2rems } from './tools/functions'
