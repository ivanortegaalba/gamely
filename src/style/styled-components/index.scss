@import '~sanitize.css';

// Settings
@import 'settings/colors';
@import 'settings/fonts';

// Tools
@import 'tools/functions';

// Components
@import 'components/button';
@import 'components/input';
@import 'components/badge';
@import 'components/table-list';
@import 'components/form-control';
@import 'components/header';

// Elements
@import 'elements/headings';

// Objects
@import 'objects/grid';
@import 'objects/spacing';
@import 'objects/flex';

body {
  font-family: $base-font-family;
  -webkit-font-smoothing: antialiased;
}
