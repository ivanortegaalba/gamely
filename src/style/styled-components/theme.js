import { colors } from './settings/colors'
import { fonts } from './settings/fonts'
import { spacing } from './settings/spacing'

export const theme = {
  fonts: {...fonts},
  palette: {...colors},
  spacing: {...spacing}
}
