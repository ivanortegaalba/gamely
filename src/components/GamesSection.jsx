import React, { Component } from 'react'
import PropTypes from '../PropTypes'
import { Button, Header } from '../style/styled-components'
import GameList from './GameList'

const ButtonRight = Button.extend`
  float: right;
`

export class GamesSection extends Component {
  constructor (props) {
    super(props)
    this.state = {
      open: props.defaultOpen
    }
  }

  onClickSeeMore = () => {
    this.setState({ open: !this.state.open })
  }

  render () {
    const { title, games, filterGamesStartWith, ...others } = this.props
    const filteredGames = filterGamesStartWith
      ? games.filter(game => game.name.toLowerCase().startsWith(filterGamesStartWith.toLowerCase()))
      : [...games]

    return (
      <section>
        <Header palette={'light'}>
          <h1>{title}</h1>
        </Header>
        <ButtonRight
          dataSpec={'see-more-button'}
          onClick={this.onClickSeeMore}
          margin={'m'}
          marginBottom={'0'}>
          {this.state.open ? 'Show less' : 'Show all'}
        </ButtonRight>
        <GameList open={this.state.open} games={filteredGames} {...others} />
      </section>
    )
  }

}

GamesSection.propTypes = {
  title: PropTypes.string,
  games: PropTypes.arrayOf(PropTypes.game),
  addToFavourites: PropTypes.func.isRequired,
  removeFromFavourites: PropTypes.func.isRequired,
  defaultOpen: PropTypes.bool,
  filterGamesStartWith: PropTypes.string
}

GamesSection.defaultProps = {
  title: '',
  defaultOpen: false,
  games: [],
  filterGamesStartWith: ''
}
