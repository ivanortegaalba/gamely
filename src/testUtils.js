import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import { ThemeProvider } from 'styled-components'



export const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}
