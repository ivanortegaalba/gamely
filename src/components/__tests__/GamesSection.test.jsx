import React from 'react'
import mockGame from '../../__mocks__/game'
import { shallowWithTheme } from '../../testUtils'
import { theme } from '../../style/styled-components/theme'
import { GamesSection } from '../GamesSection'
import { GameList } from '../GameList'

describe('<GamesSection />', () => {
  describe('init', () => {
    const removeFromFavourites = jest.fn()
    const addToFavourites = jest.fn()
    const component = (
      <GamesSection
        addToFavourites={addToFavourites}
        removeFromFavourites={removeFromFavourites}
      />
    )
    const wrapper = shallowWithTheme(component, theme)
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should change the state when the show more button is clicked', () => {
      expect(wrapper.state()).toEqual({ 'open': false })
      wrapper.find({ dataSpec: 'see-more-button' }).simulate('click')
      expect(wrapper.state()).toEqual({ 'open': true })
    })
  })

  describe('has many games', () => {
    const removeFromFavourites = jest.fn()
    const addToFavourites = jest.fn()
    const games = [{ ...mockGame, isFavourite: true }, { ...mockGame }]
    const component = (
      <GamesSection
        games={games}
        removeFromFavourites={removeFromFavourites}
        addToFavourites={addToFavourites} />)
    const wrapper = shallowWithTheme(component, theme)
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should pass the games to GameList correctly', () => {
      expect(wrapper.find('GameList').props()['games']).toEqual(games)
    })
    it('should filter games before to pass them to GameList', () => {
      const filterGamesStartWith = 'a'
      const expectedGame = { ...mockGame, name: 'a' }
      const games = [expectedGame, { ...mockGame, name: 'b' }]
      wrapper.setProps({ filterGamesStartWith, games })
      wrapper.update()
      expect(wrapper.find('GameList').props()['games']).toEqual([expectedGame])
    })
  })
})
