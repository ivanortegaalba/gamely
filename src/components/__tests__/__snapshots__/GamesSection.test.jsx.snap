// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamesSection /> has many games should render correctly 1`] = `
<section>
  <styled.header
    palette="light"
  >
    <h1 />
  </styled.header>
  <styled.button
    dataSpec="see-more-button"
    margin="m"
    onClick={[Function]}
    palette="default"
    reverse={false}
    size="m"
  >
    Show all
  </styled.button>
  <GameList
    addToFavourites={[Function]}
    defaultOpen={false}
    games={
      Array [
        Object {
          "hasBoosters": false,
          "isFavourite": true,
          "name": "8-Ball Pool",
          "short": "eightballpool",
          "tags": "",
          "url": "/spel/sportspel/8-ball-pool/?language=sv",
        },
        Object {
          "hasBoosters": false,
          "name": "8-Ball Pool",
          "short": "eightballpool",
          "tags": "",
          "url": "/spel/sportspel/8-ball-pool/?language=sv",
        },
      ]
    }
    open={false}
    removeFromFavourites={[Function]}
  />
</section>
`;

exports[`<GamesSection /> init should render correctly 1`] = `
<section>
  <styled.header
    palette="light"
  >
    <h1 />
  </styled.header>
  <styled.button
    dataSpec="see-more-button"
    margin="m"
    onClick={[Function]}
    palette="default"
    reverse={false}
    size="m"
  >
    Show all
  </styled.button>
  <GameList
    addToFavourites={[Function]}
    defaultOpen={false}
    games={Array []}
    open={false}
    removeFromFavourites={[Function]}
  />
</section>
`;
